uber/site_sections/jobs.py

```diff
@@ -45,100 +53,51 @@ def index(self, session, location=None, message=''):
         }
 
-    @ajax
-    def assign_from_everywhere(self, session, job_id, staffer_id):
-        message = session.assign(staffer_id, job_id)
-        if message:
-            return {'error': message}
-        else:
-            return job_dict(session.job(job_id))
-
-    @ajax
-    def unassign_from_everywhere(self, session, id):
-        try:
-            shift = session.shift(id)
-            session.delete(shift)
-            session.commit()
-        except:
-            return {'error': 'Shift was already deleted'}
-        else:
-            return job_dict(session.job(shift.job_id))
-
-    @ajax
-    def set_worked_from_everywhere(self, session, id, status):
-        try:
-            shift = session.shift(id)
-            shift.worked = int(status)
-            session.commit()
-        except:
-            return {'error': 'Unexpected error setting status'}
-        else:
-            return job_dict(session.job(shift.job_id))
-
     def staffers(self, session, location=None, message=''):
-        attendee = session.admin_attendee()
-        if location == 'All':
-            location = None
-        else:
-            location = int(location or c.JOB_LOCATION_OPTS[0][0])
-        jobs, shifts, attendees = session.everything(location)
-        if location:
-            attendees = [a for a in attendees if a.assigned_to(location)]
-        hours_here = defaultdict(int)
-        for shift in shifts:
-            hours_here[shift.attendee] += shift.job.weighted_hours
+        location = None if location == 'All' else int(location or c.JOB_LOCATION_OPTS[0][0])
+        attendees = session.staffers().filter(*[Attendee.assigned_depts.contains(str(location))] if location else []).all()
         for attendee in attendees:
-            attendee.hours_here = hours_here[attendee]
             attendee.trusted_here = attendee.trusted_in(location) if location else attendee.trusted_somewhere
+            attendee.hours_here = sum(shift.job.weighted_hours for shift in attendee.shifts) if location else attendee.weighted_hours
+
+        counts = defaultdict(int)
+        for job in session.jobs(location):
+            update_counts(job, counts)
+
         return {
-            'location':           location,
-            'attendees':          attendees,
-            'checklist':          session.checklist_status('assigned_volunteers', location),
-            'emails':             ','.join(a.email for a in attendees),
-            'regular_total':      sum(j.total_hours for j in jobs if not j.restricted),
-            'restricted_total':   sum(j.total_hours for j in jobs if j.restricted),
-            'all_total':          sum(j.total_hours for j in jobs),
-            'regular_signups':    sum(s.job.weighted_hours for s in shifts if not s.job.restricted),
-            'restricted_signups': sum(s.job.weighted_hours for s in shifts if s.job.restricted),
-            'all_signups':        sum(s.job.weighted_hours for s in shifts)
```

Comment:
The refactor here is instead of iterating over all jobs for every one of these statistics, we iterate over them once and then call the new ``update_counts`` function.  Not quite as clean, but not too bad and definitely improves performance.